libs/hiraya/src/components/Button/style.ts

```diff
@@ -9,30 +14,27 @@ interface ButtonProps {
   buttonSizes?: keyof typeof buttonSizeStyle;
   isFullWidth?: boolean;
   icon?: ReactNode;
+  disabled?: boolean;
 }
 
 const ButtonStyled = styled(MUIButton)<ButtonProps>(
-  ({ variants, buttonSizes, isFullWidth, icon }) => ({
-    backgroundColor: variantStyle[variants || 'primary'].background,
-    color: variantStyle[variants || 'primary'].text,
-    padding: '0',
+  ({ variants = 'primary', buttonSizes, isFullWidth, icon, disabled }) => ({
+    ...buttonSizeStyle[buttonSizes || 'large'],
+    ...focusStyle(variants),
 
     '&.MuiButton-root': {
-      minWidth: icon
-        ? buttonSizeStyle[buttonSizes || 'large']
-        : isFullWidth
-        ? '112px'
-        : '92px',
-      height: buttonSizeStyle[buttonSizes || 'large'],
+      minWidth: icon && '48px',
+      width: isFullWidth ? '100%' : buttonSizeStyle[buttonSizes || 'large'],
     },
 
+    ...(disabled
+      ? { ...disabledState }
+      : { ...variantStyle[variants || 'primary'] }),
```

Comment:
...{ variantStyle[variants || 'primary'], focusStyle(variants) },

I thought you were gonna do it like this, but this is good as well, great job!